### ISSUE TITLE
Failing fib_lookup on 5.3 kernel

### ISSUE THREAD

**Issue by brb** [cilium/cilium#9786]
A user reported that `fib_lookup` were failing on the 5.3 kernel, although `ip neigh` entry was present. E.g.:

```
$ cilium monitor -t drop
xx drop (FIB lookup failed) flow $FLOW to endpoint 0, identity 0->0: $IP1 -> $IP2

$ ip neigh
$IP2 dev ens5 lladdr $MAC2 PERMANENT
```

After the user had switched to the 5.0 kernel, the lookups stopped failing.

It would be useful to know, whether `ip route get $IP2 from $IP1` returned a route via `ens5`. If not, then the failure was expected.

/cc @jrfastab @lbernail

**Comment by Frankkkkk** [comment]
Hi.
I think we have the same problem.
For example, if we expose the kubernetes service to the outside:

```
root@pik-c1-11:~# kl get svc | grep fff
kubernetes   LoadBalancer   2001:1600:cafe:1001::1   2001:1600:cafe:1001:ffff::   443:31122/TCP   3h8m
root@pik-c1-11:~# klks exec -ti cilium-wpbd5 cilium service list | grep ffff
8    [2001:1600:cafe:1001:ffff::]:443      LoadBalancer   1 => [2001:1600:cafe:1001:1::]:6443       
````
a `curl https://[2001:1600:cafe:1001:1::]:6443` works from the node.

However, from the outside, using the external ip:
```
xx drop (FIB lookup failed) flow 0x2ec914bd to endpoint 0, identity 0->0: [2001:1600:cafe:1001:1::]:36994 -> [2001:1600:cafe:1001:1::]:6443 tcp SYN
```

**Comment by brb** [comment]
@Frankkkkk Have you redacted the source IP addr in the `drop` output? Was the kube-apiserver running on the same node which received the request?

**Comment by brb** [comment]
`[2001:1600:cafe:1001:1::]:36994 -> [2001:1600:cafe:1001:1::]:6443` - I'm confused by this output. The src and dst IP addrs are the same. Was this request sent from inside the node?

**Comment by Frankkkkk** [comment]
Hi,
The request was made from the outside ;-)
The cilium cm is :
```yaml
  auto-direct-node-routes: "false"
  bpf-ct-global-any-max: "262144"
  bpf-ct-global-tcp-max: "524288"
  cluster-name: default
  debug: "false"
  device: enp4s0f0.bgp
  enable-endpoint-routes: "true"
  enable-external-ips: "true"
  enable-host-reachable-services: "true"
  enable-ipv4: "false"
  enable-ipv6: "true"
  enable-node-port: "false"
  enable-remote-node-identity: "true"
  enable-well-known-identities: "false"
  identity-allocation-mode: crd
  install-iptables-rules: "true"
  kube-proxy-replacement: strict
  masquerade: "false"
  monitor-aggregation: low
  monitor-aggregation-flags: all
  monitor-aggregation-interval: 5s
  preallocate-bpf-maps: "false"
  sidecar-istio-proxy-image: cilium/istio_proxy
  tofqdns-enable-poller: "false"
  tunnel: disabled
  wait-bpf-mount: "false"
```
And the external ip was given by metallb running in layer2 mode (as the ip is already announced by the node).
Cheers

**Comment by brb** [comment]
Thanks. What was the client IP addr? Also, can you dump `bpftool map dump pinned /sys/fs/bpf/tc/globals/cilium_nodeport_neigh6` (you can send it to me in private)?

**Comment by Frankkkkk** [comment]
```
root@pik-c1-11:~# klks exec -ti  cilium-4qmgd bpftool map dump pinned /sys/fs/bpf/tc/globals/cilium_nodeport_neigh6
key: 20 01 16 00 00 00 cc cc  f0 00 00 00 00 00 01 86  value: b8 59 9f 6c c6 65 10 01
Found 1 element
```
Client ip is `2001:1600:0:cccc:f000::186`

**Comment by brb** [comment]
Discussed offline - this is a different problem than in the issue description.

**Comment by brb** [comment]
OT: @lbernail I've noticed that Ubuntu 18.04 LTS 5.0 kernel has backported all patches required for the host-reachable services.

**Comment by lbernail** [comment]
@brb this is really good news! We're currently trying to standardize on 20.04 to use the most recent LTS but this will take a while of course

**Comment by jrfastab** [comment]
Local testing/repro resolved by commit `86858947d0e52 ("cilium: encrypt-node creates two IPsec tunnels but only uses one")`
